src/lib.rs

```diff
@@ -240,6 +297,22 @@ impl UinputAbsSetup {
     }
 }
 
+/// The common trait for all [`InputEvent`] variants and the `InputEvent` itself.
+/// Anything that implements this can be sent to a [`Device`] or [`uinput::VirtualDevice`]
+pub trait EvdevEvent: AsRef<input_event> {
```

Comment:
if `evdev` is the crate name, that generally means to me that the symbol `evdev` shouldn't show up in names in the API.... because the context makes it clear!

what are other good names for this trait... maybe `EventWrapper` or `EventData`?

Reply:
I think `EventData` is good, given that the trait has the `AsRef<input_event>` bound.